scripts/build.mjs

```diff
@@ -15,34 +16,56 @@ if (!jobs) {
   }
 }
 
-try {
-  await which("emcmake");
-} catch {
-  console.log("ERROR: Couldn't find `emcmake`, is emsdk installed?");
-  process.exit(1);
-}
+async function build() {
+  try {
+    await which("emcmake");
+  } catch {
+    console.log("ERROR: Couldn't find `emcmake`, is emsdk installed?");
+    process.exit(1);
+  }
+
+  // Create build folder if needed
+  if (!(await fs.exists("./build"))) {
+    await $`mkdir -p build`;
+  }
+
+  cd("build");
 
-// Create build folder if needed
-if (!(await fs.exists("./build"))) {
-  await $`mkdir build`;
+  // Build with emsdk
+  await $`emcmake cmake -G Ninja ..`;
+  await $`cmake --build . -j ${jobs}`;
+
+  // Bundle api.js and copy artifacts to dist
+  await fs.copy("../src/api.js", "api.js");
+  await $`esbuild api.js --bundle --platform=node --outfile=../dist/api.js`;
+  await fs.copy("../src/cli.js", "../dist/cli.js");
+  await fs.copy("../postject-api.h", "../dist/postject-api.h");
+
+  cd("..");
```

Comment:
didn't expect the directory change to persist after exiting the function scope :/ a `defer` instruction would be useful here